app/services/file-embeddings.ts

```diff
@@ -0,0 +1,384 @@
+import { Octokit } from '@octokit/rest';
+import { supabase } from '../../src/lib/supabase';
+import { Repository } from '../types/github';
+import { createHash } from 'crypto';
+
+// Import the existing embedding service
+import { generateEmbedding } from '../services/embeddings';
+
+interface FileEmbedding {
+  repository_id: string;
+  file_path: string;
+  embedding: number[];
+  content_hash: string;
+  last_indexed_at: string;
+}
+
+/**
+ * Generate embeddings for repository files
+ */
+export async function generateFileEmbeddings(
+  repository: Repository,
+  octokit: Octokit,
+  filePaths: string[]
+): Promise<void> {
+  console.log(`Generating embeddings for ${filePaths.length} files in ${repository.full_name}`);
+  
+  try {
+    // Get repository record
+    const { data: dbRepo } = await supabase
+      .from('repositories')
+      .select('id')
+      .eq('github_id', repository.id)
+      .single();
+    
+    if (!dbRepo) {
+      console.error('Repository not found in database');
+      return;
+    }
+    
+    const embeddings: FileEmbedding[] = [];
```

Comment:
refactor (suggestion): Consider processing and inserting embeddings in smaller chunks rather than accumulating all in memory as you could potentially run out of memory if there are a large number of embeddings.